__tests__/app.test.js

```diff
@@ -27,174 +27,280 @@ describe("GET /api/topics", () => {
+    };
+    return request(app)
+      .post("/api/articles/100/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("not found");
+      });
+  });
+  test("400: responds error message of 'bad request' if article_id is wrong type", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "best article ever",
+    };
+    return request(app)
+      .post("/api/articles/cheese/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
+      });
+  });
+  test("400: responds error message of 'user does not exist' if user doesn't exist", () => {
+    const newComment = {
+      username: "vh232",
+      body: "best article ever",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("user does not exist");
+      });
+  });
+  test("400: responds error message of 'incorrect format' if input comment has incorrect key", () => {
+    const newComment = {
+      usernames: "butter_bridge",
+      body: "best article ever",
+    };
+    const newComment2 = {
+      username: 'butter_bridge',
+      body: "best article ever",
+      extraKey: "whoops"
+    }
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("incorrect format");
       })
-    })
   });
-  test('200: returns array of article objects arranged by date in ascending order', () => {
+  test("400: responds error message of 'incorrect format' if input comment doesn't contain all keys needed", () => {
+    const newComment = {
+      username: "butter_bridge",
+    };
     return request(app)
-    .get("/api/articles")
-    .expect(200)
-    .then(( { body }) => {
-      const { articles } = body
-      expect(articles).toBeSortedBy('created_at', { descending: false})
-    })
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("incorrect format");
+      })
   });
-  test('200: returns array of article objects without a "body" property', () => {
+  test("400: responds error message of 'incorrect format' if input comment has extra keys", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "best article ever",
+      extraKey: "whoops"
+    };
     return request(app)
-    .get("/api/articles")
-    .expect(200)
-    .then(( { body }) => {
-      const { articles } = body
-      expect(articles.length).toBe(13)
-      articles.forEach((article) => {
-        expect(article).not.toHaveProperty('body')
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("incorrect format");
       })
```

Comment:
This is a very important test to have, but should this return an error if given extra unnecessary keys? Is it incorrect to give extra keys in this case, as long as the necessary keys are given and they are valid?